File: example/Apollo.py
from dbassistant.interfaces.IDBConnection import IDBConnection 

class Apollo(IDBConnection):

    def connect(self):
        pass

    def disconnect(self):
        pass

    def query(self, graphql_query):
        import requests
        import json

        url = 'http://localhost:4000/'
        headers = {'Content-Type': 'application/json'}
        query = "query { " + graphql_query + " }"
        response = requests.post(url, json={'query': query}, headers=headers)
        return json.loads(response.text)

File: example/apollo.ts
import {IDBConnection} from "data-assistant";

export default class Apollo implements IDBConnection {
    connect(): void {
        // Implement connect method
    }

    disconnect(): void {
        // Implement disconnect method
    }

    query(graphqlQuery: string): any {
        const url = 'http://localhost:4000/';
        const headers = { 'Content-Type': 'application/json' };
        const query = `query { ${graphqlQuery} }`;

        const response = fetch(url, {
            method: 'POST',
            headers: headers,
            body: JSON.stringify({ query: query })
        }).then(response => response.json());

        return response;
    }
}